tests/integration/devfile/cmd_devfile_list_test.go

```diff
@@ -0,0 +1,79 @@
+package devfile
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/openshift/odo/tests/helper"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("odo list with devfile", func() {
+	var cmpName string
+	var commonVar helper.CommonVar
+
+	// This is run before every Spec (It)
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		cmpName = helper.RandString(6)
+		helper.Chdir(commonVar.Context)
+	})
+
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	When("two components in different applications", func() {
+		It("should correctly output component information", func() {
```

Comment:
Wondering if we really need that big `It` block. Can't we separate the setup into a `BeforeEach`?

Reply:
https://github.com/openshift/odo/pull/4820#issuecomment-866813087